### PR TITLE
Add __delattr__ to _ModuleWithDeprecations.

### LINKED ISSUE #2844: _ModuleWithDeprecations doesn't handle patching properly.
`_ModuleWithDeprecations` catches `__getattr__` and `__setattr__` to patch through to the underlying module, but does not intercept `__delattr__`. That means that if you're using something like `mock.patch`, the mock successfully lands in place, but cannot be removed: the mock was applied to the underlying module, but the delete comes from the proxy.

Should be easily fixed.

### PR DESCRIPTION
**Pull request by Lukasa**
Resolves #2844.

### CODE REVIEW

**Comment by Lukasa** [comment]
Test failure is spurious.

File: src/cryptography/utils.py [hunk 1]
```diff
@@ -119,6 +119,9 @@ def __getattr__(self, attr):
     def __setattr__(self, attr, value):
         setattr(self._module, attr, value)
 
+    def __delattr__(self, attr):
+        delattr(self._module, attr)
```
Comment (alex) [inline src/cryptography/utils.py]: This should raise a warning if the attribute is deprecated, since deleting it won't work shortly.

**Comment by alex** [comment]
Needs tests.

**Comment by reaperhulk** [comment]
:+1: thanks @Lukasa

**Pull request merged**